setup/src/Magento/Setup/Module/Di/Code/Reader/ClassesScanner.php

```diff
@@ -16,11 +17,29 @@ class ClassesScanner implements ClassesScannerInterface
     protected $excludePatterns = [];
 
     /**
+     * @var array
+     */
+
+    private $fileResults = [];
+
+    /**
+     * @var string
+     */
+
+    private $generationDirectory;
+
+    /**
      * @param array $excludePatterns
+     * @param string $generationDirectory
      */
-    public function __construct(array $excludePatterns = [])
+
+    public function __construct(array $excludePatterns = [], DirectoryList $directoryList = null)
     {
         $this->excludePatterns = $excludePatterns;
+        if (!$directoryList instanceof DirectoryList) {
```

Comment:
it's better to compare with null than to make instanceof opeartion.
It's better from teh performance point of view